src/providers/mssql/qgsmssqlprovider.cpp

```diff
@@ -962,13 +994,16 @@ void QgsMssqlProvider::UpdateStatistics( bool estimate ) const
       mSRId = mParser.GetSRSId();
     }
   }
+
+  return;
 }
 
 // Return the extent of the layer
 QgsRectangle QgsMssqlProvider::extent() const
 {
+  QgsError error;
   if ( mExtent.isEmpty() )
-    UpdateStatistics( mUseEstimatedMetadata );
+    UpdateStatistics( mUseEstimatedMetadata, error );
```

Comment:
Shall we use mError here like we do in createSpatialIndex() ?

Reply:
not sure, here the layer is already loaded, error is just a place holder.